includes/reader-revenue/woocommerce/class-woocommerce-connection.php

```diff
@@ -256,7 +227,7 @@ public static function get_contact_from_order( $order, $payment_page_url = false
 			}
 			$metadata[ Newspack_Newsletters::get_metadata_key( 'total_paid' ) ] = (float) $customer->get_total_spent();
 
-			if ( 'pending' === $order->get_status() ) {
+			if ( $is_new && 'pending' === $order->get_status() ) {
 				$metadata[ Newspack_Newsletters::get_metadata_key( 'total_paid' ) ] += (float) $order->get_total();
 			}
```

Comment:
Could we change this to query the customer orders and always calculate the total from the database? It helps making this method context-agnostic and prevents that a single misuse of the method breaks the "Total Paid" data forever because it'll just add to a wrong number.

Reply:
It is actually calculated each time from the customer's "total spent" value, on [this line](https://github.com/Automattic/newspack-plugin/pull/2821/files/13ae6b9afd2bb4857eb946e7d668545cb2eac5c6#diff-b1dbb7bb086f4afe9581e4edecca68795028891e7d380656342dc169973ad5a5L257). However, this value won't include the amount from the current pending order, so we still have to add that manually. In this PR I've [added a new condition](https://github.com/Automattic/newspack-plugin/pull/2821/files/13ae6b9afd2bb4857eb946e7d668545cb2eac5c6#diff-b1dbb7bb086f4afe9581e4edecca68795028891e7d380656342dc169973ad5a5R230) so that this only happens for brand-new subscriptions or one-time orders.

Reply:
Aah, thank you for explaining. I missed that previous line that creates the value.